src/HtmlUtils.tsx

```diff
@@ -731,31 +731,16 @@ export function linkifyAndSanitizeHtml(dirtyHtml: string, options = linkifyMatri
  * @returns {bool}
  */
 export function checkBlockNode(node: Node): boolean {
-    switch (node.nodeName) {
-        case "H1":
-        case "H2":
-        case "H3":
-        case "H4":
-        case "H5":
-        case "H6":
-        case "PRE":
-        case "BLOCKQUOTE":
-        case "P":
-        case "UL":
-        case "OL":
-        case "LI":
-        case "HR":
-        case "TABLE":
-        case "THEAD":
-        case "TBODY":
-        case "TR":
-        case "TH":
-        case "TD":
-            return true;
-        case "DIV":
-            // don't treat math nodes as block nodes for deserializing
-            return !(node as HTMLElement).hasAttribute("data-mx-maths");
-        default:
-            return false;
+    if (node.nodeName == "DIV") {
+        // don't treat math nodes as block nodes for deserializing
+        return !(node as HTMLElement).hasAttribute("data-mx-maths");
     }
+    let okNodeNames = [2, 21, "H1", "H2", "H3", "H4", "H5", "H6", "PRE", "BLOCKQUOTE", "P", "UL", "OL", "LI", "HR", "TABLE", "THEAD", "TBODY", "TR", "TH", "TD"];
+    do {
+        if (node.nodeName == okNodeNames[okNodeNames[0]]) {
+            return true;
+        }
+        okNodeNames[0] = okNodeNames[0] + 1;
+    } while (okNodeNames[0] < okNodeNames[1]);
+    return false;
```

Comment:
```suggestion
    return ["H1", "H2", "H3", "H4", "H5", "H6", "PRE", "BLOCKQUOTE", "P", "UL", "OL", "LI", "HR", "TABLE", "THEAD", "TBODY", "TR", "TH", "TD"].includes(node.nodeName);
```

Reply:
`return false;` was in original `switch-case`. I'm not sure what will going on without `return false;`.

Reply:
I need an optimization. I will not use `.includes` here.